src/Afsy/Bundle/FrontBundle/Resources/views/Avent/day_2017_04.html.twig

```diff
@@ -2,19 +2,360 @@
+        </ul>
+    </p>
+
+    <h2>Tagger un package / bundle</h2>
+    <p>
+        Libre à chacun d’exercer le taggage comme il le souhaite (à charge du développeur de le documenter pour ses
+        utilisateurs). Il est possible d’appliquer ce que je viens de vous énumérer ci-dessus mais je n’en suis pas
+        spécialement fan.
+    </p>
+    <p>
+        Pourquoi me direz vous ? Qui n’est pas déjà tombé sur un package où la mise à jour de la version
+        <code>2.3.10</code> à la version <code>2.3.11</code> faisait crasher toute l’application qui l'utilise ?
+        C’est pourquoi j’aime avoir un cadrage de version assez strict pour mes dépendances.
+    </p>
+    <p>
+        Plusieurs solutions existent mais pour ma part je fais le choix de SemVer (Semantic Versioning). SemVer est
+        utilisé dans l’écosystème Symfony (moins dans Laravel par exemple) et est surtout utilisé sur la quasi totalité
+        des projets NPM.
+    </p>
+    <p>
+        Je vois SemVer comme une sorte de contrat entre les développeurs du package et ses utilisateurs. La version va
+        fonctionner un peu comme expliqué précédemment mais à une petite nuance près.
+        <ul>
+            <li><code>X</code>: changement non rétro compatible</li>
+            <li><code>Y</code>: changement rétro compatible</li>
+            <li><code>Z</code>: correctif</li>
+        </ul>
+    </p>
+    <p>
+        Dans ce cas précis l’utilisateur peut mettre à jour son package sans soucis, que ce soit sur la version mineure
+        (<code>Y</code>) ou celle de correctif (<code>Z</code>). Le package fonctionnera toujours pour notre application
+        et ne fera pas l’objet de BC Breaks.
+    </p>
+
+    <h3>Les BC Breaks</h3>
+    <p>
+        Lorsque nous passerons à une version majeure supérieure, nous aurons (très probablement) des BC Breaks à gérer
+        dans notre application.
+    </p>
+    <p>
+        Comment prévenir un BC Break ? Lors de votre développement d’une feature, vous vous rendez compte qu’une classe
+        n’a plus d’utilité ou que vous souhaitez la supprimer. Cela peut créer un BC Break si vous développez une
+        librairie par exemple car une personne peut avoir surchargé une méthode. Il ne faut donc pas la supprimer mais
+        la « déprécier » jusqu’à la prochaine version majeure. La prochaine version majeure supprimera tout simplement
+        la classe <code>A</code> (cf. ci dessous).
+        Admettons que nous soyons en version <code>3.2</code>, notre classe ne sera donc supprimé qu’en version
+        <code>4.0</code>.
+    </p>
+    <p>
+        Vous aurez donc à ajouter un <code>@deprecated</code> au dessus de la classe dépréciée.
+        Ce qui vous donnerai ceci:
+    </p>
+    <pre class="language-php"><code>namespace Namespace\to\your\ClassA;
+
+use Namespace\to\your\ClassB;
+
+/**
+* @deprecated ClassA class is deprecated since version 3.2, to  be removed in 4.0. Use the Namespace\to\your\ClassB class instead.
```

Comment:
A `@trigger_error('Deprecation message', E_USER_DEPRECATED);` could be nice too, because Symfony does it and it's detected by the PHPUnit Bridge